### PR TITLE
Bug 1819970 - Add uplift process documentation

### PR DESCRIPTION
**Pull request by DonalMe**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1819970

### CODE REVIEW

File: docs/shared/uplift_guide.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# Uplift Request Process Guide
+
+An uplift can be requested of a patch landed on main in order to expedite that change into an earlier release outside of the standard train model.
+
+The developer provides some information as part of the request to help inform the decision as to whether or not to accept the uplift. These requests are regularly monitored by Release Management. The release manager will then assess the potential risk/reward and makes a decision on whether or not to accept the requested uplift. If the uplift is accepted, the release manager approves the request and grafts the patch to the Beta branch. If the request does not meet the criteria, the release manager may look for some additional information or reject the request and provide a reason.
```
Comment (dsmithpadilla) [inline docs/shared/uplift_guide.md:5]: s/makes/make

File: docs/shared/uplift_guide.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+## Beta/Release Uplift Steps
+
+For Beta and Release, the following steps outline the process:
+1. A PR that targets Firefox-Android main has an associated Bugzilla bug. The PR is linked as an attachment to the Bugzilla bug.
+2. The PR goes through the standard review process and merging into the main branch.
```
Comment (dsmithpadilla) [inline docs/shared/uplift_guide.md:11]: s/merging/merges

**Review by dsmithpadilla — APPROVED** [review]

**Comment by rvandermeulen** [comment]
@DonalMe Please either have this landed as a squashed PR or fold the follow-up commit into the first one.

File: docs/shared/uplift_guide.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    - Use Bugzilla’s “needinfo” to request engineering or product manager in the Bugzilla bug, so we have a record of the approvals.
+    - Some features and technical changes may need decisions and guidance from the leadership team that can affect what is released to our users in the Beta and/or Production channels.
+    - Uplifting small bug fixes doesn’t require approval from an engineering or product manager.
+- An engineering or product manager may identify that a PR is applicable for an uplift request and reach out to the developer.
+- The autonag bot may identify that a bug was fixed in main but Beta or Release are marked as affected. The bot will add a comment to the bug and needinfo the developer.
+- Release Management may identify if a PR is a good candidate to backport and reach out to the developer via a comment in the bug.
+4. If there is agreement and manager approval that the patch is suitable for uplift, the developer uses Mergify to create a [backport PR](https://docs.mergify.com/actions/backport/) for the applicable release branch.
+- Example: `@mergify backport releases_v111`
```
Comment (rvandermeulen) [inline docs/shared/uplift_guide.md:21]: Shouldn't this be `@Mergifyio`?

**Comment by DonalMe** [comment]
@Mergifyio squash

**Review by rvandermeulen — APPROVED** [review]

**Pull request merged**